src/rtk/features/delegableApi/delegableApiSlice.ts

```diff
@@ -47,24 +69,82 @@ const delegableApiSlice = createSlice({
   name: 'delegableApi',
   initialState,
   reducers: {
-    softAdd: (state, action) => {
+    softAdd: (state: SliceState, action: any) => {
       const { delegableApiList } = state;
+      const { presentedApiList } = state;
+      const { delegableApiSearchPool } = state;
       state.delegableApiList = delegableApiList.filter(
         (delegableApi) => delegableApi.id !== action.payload.id,
       );
+      state.presentedApiList = presentedApiList.filter(
+        (delegableApi) => delegableApi.id !== action.payload.id,
+      );
+      state.delegableApiSearchPool = delegableApiSearchPool.filter(
+        (delegableApi) => delegableApi.id !== action.payload.id,
+      );
 
       state.chosenDelegableApiList.push(action.payload);
     },
-    softRemove: (state, action) => {
+    softRemove: (state: SliceState, action: any) => {
       state.delegableApiList.push(action.payload);
+      state.presentedApiList.push(action.payload);
+      state.delegableApiSearchPool.push(action.payload);
 
       const { chosenDelegableApiList } = state;
       state.chosenDelegableApiList = chosenDelegableApiList.filter(
         (delegableApi) => delegableApi.id !== action.payload.id,
       );
     },
+    filter: (state: SliceState, action: any) => {
+      const { delegableApiList } = state;
+      const filterList = action.payload;
+      let searchPool = [...delegableApiList];
+      if (filterList && filterList.length > 0) {
+        searchPool = delegableApiList.filter((api) =>
+          filterList.some((filter: string) => {
+            if (api.orgName.toLocaleLowerCase().includes(filter.toLowerCase())) {
+              return true;
+            }
+            return false;
+          }),
+        );
+        state.delegableApiSearchPool = searchPool;
+      } else {
+        state.delegableApiSearchPool = delegableApiList;
+      }
+    },
+    search: (state: SliceState, action: any) => {
+      const { delegableApiSearchPool } = state;
+      const searchText = action.payload.trim().toLowerCase();
+      const seachWords = searchText ? searchText.split(' ') : [];
+
+      const prioritizedApiList: DelegableApiWithPriority[] = [];
+      if (searchText) {
+        for (const api of delegableApiSearchPool) {
+          let numMatches = 0;
+          for (const word of seachWords) {
+            if (
+              api.apiName.toLowerCase().includes(word) ||
+              api.description.toLowerCase().includes(word) ||
+              api.orgName.toLowerCase().includes(word)
+            ) {
+              numMatches++;
+            }
+          }
+          if (numMatches > 0) {
+            prioritizedApiList.push({ ...api, priority: numMatches });
+          }
+        }
+
+        state.presentedApiList = prioritizedApiList
+          .sort((a, b) => (a.priority < b.priority ? 1 : -1))
+          .map(({ priority, ...otherAttr }) => otherAttr);
```

Comment:
Does the map do anything?

Reply:
It removes the priority prop from the array. There might be a better way to do that though - I'm open to suggestions 🙂
Or maybe removing it isn't necessary at all? What do you think?

Reply:
I see, I guess the priority prop isn't displayed in the AccordionContent, if so you don't need to remove the prop and can remove the line code line with map 😄

Reply:
You know what, as I look at it I see that if I remove it, I'll have to add the priority prop to state.presentedApiList and we'll have to maintain it (on adding or removing from the list( and I'll have to add a map two lines below this one anyway (`state.presentedApiList = delegableApiSearchPool;` will need mapping to include the priority).

... or I guess I can make priority optional...
It might look weird in the store for a sec, for some to have priorities and some to not, but we trigger a new search right after adding anyway so it won't be for long. The larger problem with this approach is probably that it will look like an odd choice to have a different type on the presented list, one with an optional priority value that is never used. That might cause some head-scratches in the readability department.

Reply:
Seems like keeping the map function is the most trivial, but you're free to choose as I don't have enought knowledge about this.

Reply:
You can resolve this conversation when you've decided what you wanna do 😊

Reply:
I'll just leave it. :P